### ISSUE TITLE
Move codebase to non-nullablity

### ISSUE THREAD

**Issue by happy-san** [typesense/typesense-dart#37]
## Description
According to the [docs ](https://dart.dev/null-safety/migration-guide#step1-wait)

>You can — and should — migrate your package before packages that depend on it are migrated. Your null-safe package is usable by packages and apps that don’t use null safety yet, as long as they use Dart 2.12 or later. For example, the Dart and Flutter core libraries are null safe, and they’re still usable by apps that haven’t migrated to null safety.

Since it's encouraged to be in Dart 2.12 or later, we'll update the dart client.

**Comment by happy-san** [comment]
[welp](https://github.com/dart-lang/sdk/issues/46350)

**Comment by happy-san** [comment]
Preview of the null safe package is available [here](https://github.com/happy-san/typesense-dart/tree/nnbd_migration).

To use the pre-release version, add this dependency in `pubspec.yaml`:
```yaml
  typesense:
    git:
      url: https://github.com/happy-san/typesense-dart.git
      ref: nnbd_migration
```

**Comment by jasonbosco** [comment]
@hitiksaini Please see @happy-san's comment above about null safety pre-release version ^

**Comment by hitiksaini** [comment]
> @hitiksaini Please see @happy-san's comment above about null safety pre-release version ^

Thank you!! @happy-san @jasonbosco 🙌 

It is compatible with new null safe environment in our project. A big thanks to you guys from my team 👏😍

**Comment by happy-san** [comment]
@hitiksaini this feature landed in typesense 0.2.0
https://pub.dev/packages/typesense
